### PR TITLE
Compute the queue level of driver by time slice

### PR DESCRIPTION
**Pull request by ZiheLiu**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] others

## Problem Summary：
The queue level of driver is determined by the times it runs on the driver worker thread. That is , `level=log2(times)`.

However, if the driver costs only a little time for multiple times, it will move to the low-priority level queue quickly, although it only consumes a little time.

Therefore, we could determine the queue level of a driver by the driver consuming time instead of execution times. The time slice of the `i-th` level for a driver is `(i+1)*200`ms. That is, when a driver's execution time exceeds 0.2s, 0.6s, 1.2s, 2s, 3s, 4.2s, 5.6s, and 7.2s, it will move to next level.

### CODE REVIEW

File: be/src/exec/pipeline/pipeline_driver_queue.h [hunk 1]
```diff
@@ -123,11 +147,21 @@ class QuerySharedDriverQueueWithoutLock : public FactoryMethod<DriverQueue, Quer
...
+    // it will move to (i+1)-th level.
+    int _compute_driver_level(const DriverRawPtr driver) const;
 
+private:
     static constexpr size_t QUEUE_SIZE = 8;
     static constexpr double RATIO_OF_ADJACENT_QUEUE = 1.2;
+    // The time slice of the i-th level is (i+1)*LEVEL_TIME_SLICE_BASE ns,
+    // so when a driver's execution time exceeds 0.2s, 0.6s, 1.2s, 2s, 3s, 4.2s, 5.6s, and 7.2s,
+    // it will move to next level.
+    static constexpr int64_t LEVEL_TIME_SLICE_BASE_NS = 200'000'000L;
```
Comment (kangkaisen) [inline be/src/exec/pipeline/pipeline_driver_queue.h:160]: put to parent  class？

Comment (kangkaisen) [inline be/src/exec/pipeline/pipeline_driver_queue.h:160]: why we choose 200ms?

**Comment by kangkaisen** [comment]
What's the optimized result of this PR?

**Review by satanson — DISMISSED** [review]

File: be/src/exec/pipeline/pipeline_driver_queue.h (hunk 1 above)
Comment (ZiheLiu) [reply be/src/exec/pipeline/pipeline_driver_queue.h:160]: It's just an empirical value. The yield time of driver is 100ms, so maybe let a driver run 2 times on the highest priority queue.

**Comment by ZiheLiu** [comment]
> What's the optimized result of this PR?

@kangkaisen After this improvement, the cpu usage of the 64 concurrency query becomes stable, but the total query time only improve very little.

File: be/src/exec/pipeline/pipeline_driver_queue.h (hunk 1 above)
Comment (ZiheLiu) [reply be/src/exec/pipeline/pipeline_driver_queue.h:160]: OK.

Comment (mofeiatwork) [reply be/src/exec/pipeline/pipeline_driver_queue.h:160]: Will it overfit to this workload, and regressed in other workloads?

Comment (ZiheLiu) [reply be/src/exec/pipeline/pipeline_driver_queue.h:160]: @mofeiatwork  The yield time of driver is fixed 100ms for all the scenarios. Let me do most benchmark tests on SSB, TPC-DS, and TPC-H dataset.

**Comment by ZiheLiu** [comment]
run starrocks_clang-format

**Review by kangkaisen — APPROVED** [review]

**Review by satanson — APPROVED** [review]

**Pull request merged**